forte/evaluation/base/base_evaluator.py

```diff
@@ -93,6 +94,32 @@ def ref_pack_record(self, record_meta: Dict[str, Set[str]]):
         """
         pass
 
+    def collect_input_pack_record(self,
+                                  input_pack: PackType) -> Dict[str, Set[str]]:
+        # pylint: disable=protected-access
+        r"""Method to collect the type and attributes from the input pack and if
+        :attr:`~forte.pipeline.Pipeline.resource` has `onto_specs` as key
+        and ontology specification file path as value, then
+        `merged_entry_tree` that has all the entries in ontology specification
+        file would be populated. All the parent entry nodes of the input pack
+        would be collected from this tree and add to the returned record
+        dictionary for later comparison to enable subclass type checking.
+
+        Args:
+            input_pack: The input datapack.
+
+        Returns:
+            input_pack_record: The input pack record content combined with
+            all the parent types and attributes collected from
+            merged_entry_tree
+
+        """
+        input_pack_record = input_pack._meta.record.copy()
```

Comment:
Is this function used twice or more? We should create a small util function and reference it, don't copy the code.